src/cluster_legacy.c

```diff
@@ -3113,6 +3113,17 @@ int clusterProcessPacket(clusterLink *link) {
         if (sender_claims_to_be_primary && sender_claimed_config_epoch > sender->configEpoch) {
             sender->configEpoch = sender_claimed_config_epoch;
             clusterDoBeforeSleep(CLUSTER_TODO_SAVE_CONFIG | CLUSTER_TODO_FSYNC_CONFIG);
+
+            if (server.cluster->failover_auth_time && sender->configEpoch == server.cluster->failover_auth_epoch) {
```

Comment:
@PingXie Not sure if i am doing it (the time or the conditions) right here.

Reply:
Yeah I think this fix makes sense. I wonder if we should check "greater than or equal to" instead? There is no way "myself" can win an election in the past.

Reply:
I am worried that when multiple shards failover, they each elect in their own epoch. In this case, the sender's epoch may be > my own failover epoch, but it is correct (different epoch). something like #1018

Reply:
> There is no way "myself" can win an election in the past.

this make sense to me, ok, i am changing it and then i will re-test it.